pymongo/asynchronous/client_bulk.py

```diff
@@ -133,7 +133,10 @@ def add_insert(self, namespace: str, document: _DocumentOut) -> None:
         validate_is_document_type("document", document)
         # Generate ObjectId client side.
         if not (isinstance(document, RawBSONDocument) or "_id" in document):
-            document["_id"] = ObjectId()
+            new_document = {"_id": ObjectId()}
+            new_document.update(document)
+            document.clear()
+            document.update(new_document)
```

Comment:
Thoughts on the perf implications of this vs ChainMap?

Reply:
Using ChainMap is cleaner, encoding to RawBSONDocuments might have additional performance costs.